src/Microsoft.ML.Core/Data/Repository.cs

```diff
@@ -113,14 +113,19 @@ internal Repository(bool needDir, IExceptionContext ectx)
             PathMap = new ConcurrentDictionary<string, string>();
             _open = new List<Entry>();
             if (needDir)
-                DirTemp = GetShortTempDir();
+                DirTemp = GetShortTempDir(ectx);
             else
                 GC.SuppressFinalize(this);
         }
 
-        private static string GetShortTempDir()
+        private static string GetShortTempDir(IExceptionContext ectx)
         {
-            var path = Path.Combine(Path.GetFullPath(Path.GetTempPath()), "ml_dotnet", Path.GetRandomFileName());
+            string path;
+            if (ectx is IHostEnvironmentInternal iHostInternal)
```

Comment:
In what circumstance would a ectx be an `IHostEnvironmentInternal `

Reply:
It actually is all the time. The only time that it _wouldn't_ be is if an end user made their own class that implements `IExceptionContext`. That's why the check is still there for the off chance that actually happens.